File: lua/diff.lua

-- checked with Lua5.1.4

dofile("onp.lua")

if #arg < 2 then
   error("few argument")
end
a = arg[1]
b = arg[2]
d = ONP.new(a, b)
d:compose()
print("editDistance:" .. d:geteditdistance())
print("LCS:" .. d:getlcs())
print("SES")
ses = d:getses()
for i=1, #ses do
   if ses[i].type == SES_COMMON then
      print("  " .. ses[i].elem)
   elseif ses[i].type == SES_DELETE then
      print("- " .. ses[i].elem)
   elseif ses[i].type == SES_ADD then
      print("+ " .. ses[i].elem)
   end
end

File: javascript/diff.js
/** 
 * This program is checked with node.js
 */ 

var onp = require('./onp.js');

var diff = new onp.Diff(process.argv[2], process.argv[3]);
diff.compose();
console.log("editdistance:" + diff.editdistance());
console.log("lcs:" + diff.getlcs());
console.log("ses");

var i   = 0;
var ses = diff.getses();
for (i=0;i<ses.length;++i) {
    if (ses[i].t === diff.SES_COMMON) {
        console.log(" " + ses[i].elem);
    } else if (ses[i].t === diff.SES_DELETE) {
        console.log("-" + ses[i].elem);
    } else if (ses[i].t === diff.SES_ADD) {
        console.log("+" + ses[i].elem);
    }
}
